### PR TITLE
Bumping 'golangci-lint' version for SRO.

### PR DESCRIPTION
**Pull request by ybettan**

### CODE REVIEW

File: clusters/app.ci/supplemental-ci-images/golangci-lint_is.yaml [hunk 1]
```diff
@@ -15,3 +15,7 @@ spec:
     from:
       kind: DockerImage
       name: "docker.io/golangci/golangci-lint:v1.37.1"
+  - name: "v1.43.0"
+    from:
+      kind: DockerImage
+      name: "docker.io/golangci/golangci-lint:v1.43.0"
```
Comment (petr-muller) [inline clusters/app.ci/supplemental-ci-images/golangci-lint_is.yaml:21]: Not here, please follow https://docs.ci.openshift.org/docs/how-tos/external-images/

File: ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ base_images:
   golangci-lint:
     name: golangci-lint
     namespace: ci
-    tag: v1.37.1
+    tag: v1.43.0
```
Comment (petr-muller) [inline ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml:5]: You can only use the image *after* it is present in the cluster.

File: clusters/app.ci/supplemental-ci-images/golangci-lint_is.yaml (hunk 1 above)
Comment (ybettan) [reply clusters/app.ci/supplemental-ci-images/golangci-lint_is.yaml:21]: Ack, will do it, thanks.

I guess that I need someone from the CI team to approve the PR?

File: ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml (hunk 2 above)
Comment (ybettan) [reply ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml:5]: Thanks, I will split the PR then.

File: clusters/app.ci/supplemental-ci-images/golangci-lint_is.yaml (hunk 1 above)
Comment (petr-muller) [reply clusters/app.ci/supplemental-ci-images/golangci-lint_is.yaml:21]: Correct

**Comment by ybettan** [comment]
This PR is rebased on top of  https://github.com/openshift/release/pull/24526 and depends on it.

**Comment by ybettan** [comment]
```
could not resolve inputs: could not determine inputs for step [input:golangci-lint]: could not resolve base image from ci/golangci-lint:v1.43.0: imagestreamtags.image.openshift.io "golangci-lint:v1.43.0" not found
```

@petr-muller looks like we broke the images.

**Comment by petr-muller** [comment]
The images are imported hourly and I see them imported just fine.

**Pull request closed without merging**